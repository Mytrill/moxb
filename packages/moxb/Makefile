all:
	cd ../.. && $(MAKE)

# this will forward any other make goal to the parent makefile
$(MAKECMDGOALS):
	cd ../.. && $(MAKE) $(MAKECMDGOALS)
