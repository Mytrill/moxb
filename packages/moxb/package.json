{
  "name": "@moxb/moxb",
  "version": "0.1.3",
  "description": "MobX data binding",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "main": "build",
  "typings": "build/",
  "files": [
    "build/**/*"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/moxb/moxb.git"
  },
  "author": "moxb (moxb.org)",
  "license": "MIT",
  "dependencies": {
    "tslib": "^1.9.3"
  },
  "peerDependencies": {
    "mobx": "^4.3.0 || ^5.0.0"
  },
  "devDependencies": {
    "@types/jest": "^23.1.5",
    "@types/meteor": "^1.4.16",
    "jest": "^23.3.0",
    "prettier": "^1.13.7",
    "ts-jest": "^23.0.0",
    "typescript": "^2.9.2"
  },
  "bugs": {
    "url": "https://github.com/moxb/moxb/issues"
  },
  "homepage": "https://github.com/moxb/moxb#readme",
  "keywords": [
    "mobx",
    "UI"
  ]
}
