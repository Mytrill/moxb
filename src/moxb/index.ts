export * from './impl';
export * from './meteor';
export * from './types';
