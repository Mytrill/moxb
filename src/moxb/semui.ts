export * from './action/ActionUi';
export * from './bind/BindUi';
export * from './bool/BoolUi';
export * from './confirm/ConfirmUi';
export * from './many-of/ManyOfUi';
export * from './modal/ModalUi';
export * from './numeric/NumericUi';
export * from './one-of/OneOfUi';
export * from './table/TablePaginationUi';
export * from './table/TableSearchUi';
export * from './table/TableUi';
export * from './text/TextUi';
