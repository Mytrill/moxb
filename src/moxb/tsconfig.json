{
    "compilerOptions": {
        "module": "commonjs",
        "target": "ES5",
        "jsx": "react",
        "lib": ["DOM","ES6","ES2015","ES2017", "DOM.Iterable","ScriptHost"],
        "importHelpers":true,
        "experimentalDecorators": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strict": true,
        "outDir": "../build",
        "declaration": true,
        "skipLibCheck": true,
        "strictNullChecks": true,
        "removeComments": false
    },
    "compilerOptions": {
        "baseUrl": "",
        "paths": {
            "*": [
                "*",
                "../../optional-modules/node_modules/*"
            ]
        }
    },
    "exclude": [
        "../node_modules"
    ]
}
