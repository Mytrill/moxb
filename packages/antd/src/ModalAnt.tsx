import { Action, Modal as MoxbModal } from '@moxb/moxb';
import { Modal } from 'antd';
import { ModalFuncProps } from 'antd/lib/modal';
import { observer } from 'mobx-react';
import * as React from 'react';

export interface BindModalAntProps<T> extends ModalFuncProps {
    operation: MoxbModal<T>;
    footer?(actions: Action[]): React.ReactNode;
}

// same sizes as semantic UI's modals
const SIZES = {
    default: '900px', // default size for ant modal: 512px
    mini: '360px',
    tiny: '540px',
    small: '720px',
    large: '1080px',
    fullscreen: '95% !important',
};

@observer
export class ModalAnt<T> extends React.Component<BindModalAntProps<T>> {
    /*
     * Ant Design has no concept for multiple actions in a modal dialog, it will always be an 'okButton' and
     * a 'cancelButton', so we will warn the user, that not more actions are available.
     * */
    render() {
        const { operation, children, footer, width, ...props } = this.props;
        const size = this.props.width || SIZES[operation.size || 'default'];

        if (footer && operation.actions) {
            return (
                <Modal
                    {...props as any}
                    visible={operation.open}
                    title={operation.header}
                    onCancel={operation.actions[0].fire}
                    footer={footer(operation.actions)}
                    width={size}
                >
                    {children}
                </Modal>
            );
        }

        if (operation.actions!.length !== 2) {
            console.warn('The modals for ant design UI components must have a fixed number of two actions.');
            return null;
        }
        return (
            <Modal
                {...props as any}
                visible={operation.open}
                onCancel={operation.actions![0].fire}
                onOk={operation.actions![1].fire}
                cancelText={operation.actions![0].label}
                okText={operation.actions![1].label}
                title={operation.header}
                width={size}
            >
                {children}
            </Modal>
        );
    }
}
