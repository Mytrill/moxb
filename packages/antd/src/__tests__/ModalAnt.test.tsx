import { ActionImpl, ModalImpl } from '@moxb/moxb';
import { shallow } from 'enzyme';
import * as React from 'react';
import { ModalAnt } from '../ModalAnt';
import { shallowMoxbToJson } from './enzymeHelper';

describe('ModalAnt', function() {
    it('should render a modal dialog by default', function() {
        const operation = new ModalImpl<any>({
            actions: () => [
                new ActionImpl({
                    id: 'Id.btn',
                    label: 'Action',
                    fire: jest.fn(),
                }),
                new ActionImpl({
                    id: 'Id.btn1',
                    label: 'Cancel',
                    fire: jest.fn(),
                }),
            ],
            header: () => 'New Modal Header',
        });
        expect(shallowMoxbToJson(shallow(<ModalAnt operation={operation} />))).toMatchSnapshot();
    });

    it('should return null and print a warning, if only one button was defined', function() {
        const operation = new ModalImpl<any>({
            actions: () => [
                new ActionImpl({
                    id: 'Id.btn',
                    label: 'Action',
                    fire: jest.fn(),
                }),
            ],
            header: () => 'New Modal Header',
        });
        const consoleWarn = jest.spyOn(console, 'warn');
        consoleWarn.mockImplementation(() => {});
        expect(shallow(<ModalAnt operation={operation} />).type()).toBeNull();
        expect(consoleWarn).toHaveBeenCalledTimes(1);
        consoleWarn.mockRestore();
    });
});
