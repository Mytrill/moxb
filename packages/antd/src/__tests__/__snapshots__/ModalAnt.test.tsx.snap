// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalAnt should render a modal dialog by default 1`] = `
<Modal
  cancelButtonDisabled={false}
  cancelText="Action"
  confirmLoading={false}
  maskTransitionName="fade"
  okButtonDisabled={false}
  okText="Cancel"
  okType="primary"
  onCancel={[Function]}
  onOk={[Function]}
  title=""
  transitionName="zoom"
  visible={false}
  width={520}
/>
`;
